### DISCUSSION TITLE
Accuracy high, but bad confusion matrix

### DISCUSSION THREAD

**Discussion by ammarsaf** [mrdbourke/pytorch-deep-learning#229] [Q&A] [answered]
I have trained the model followed using you code 

```
from torch.nn.modules.pooling import MaxPool2d
from torch import nn
class FashionModelConv(nn.Module):
    """
    Replicate TinyVGG pre-trained model (such VGG16, ResNet, MobileNet)
    """
    def __init__(self, input_channel, hidden_units, output_shape):
        super().__init__()
        self.conv_block_1 = nn.Sequential(
                            nn.Conv2d(in_channels=input_channel, 
                                    out_channels=hidden_units, 
                                    kernel_size=3, 
                                    stride=1, 
                                    padding=1), 
                            
                            nn.ReLU(), 
                            nn.Conv2d(in_channels=hidden_units, 
                                    out_channels =hidden_units, 
                                    kernel_size=3, 
                                    stride=1, 
                                    padding=1), 

                            nn.ReLU(), 
                            nn.MaxPool2d(kernel_size=2)
                            
                        )
        self.conv_block_2 = nn.Sequential(
                            nn.Conv2d(in_channels=hidden_units, 
                                    out_channels =hidden_units, 
                                    kernel_size=3, 
                                    stride=1, 
                                    padding=1), 
                            
                            nn.ReLU(), 
                            nn.Conv2d(in_channels=hidden_units, 
                                    out_channels=hidden_units, 
                                    kernel_size=3, 
                                    stride=1, 
                                    padding=1), 

                            nn.ReLU(), 
                            nn.MaxPool2d(kernel_size=2)
                        )
        
        self.classifier = nn.Sequential(
                            nn.Flatten(), 
                            nn.Linear(in_features=hidden_units*7*7, ## Why 7x7? 
                                      out_features=output_shape)
                            )
        
    def forward(self, x:torch.Tensor):
        x = self.conv_block_1(x)
        #print(x.shape)
        x = self.conv_block_2(x)
        #print(x.shape)
        x = self.classifier(x)
        #print(x.shape)
        return x

```
With accuracy 
```
{'model': 'FashionModelConv',
 'model_loss': 0.28384771943092346,
 'model_acc': 89.85623003194888}
```

But when I create the confusion matrix, it showed the bad one. 

![download](https://user-images.githubusercontent.com/88831990/209054482-2767af1c-538e-4ae3-a094-eabacdc17ad8.png)

Been tried everything for hours. Anyone can help me?

**Comment by ammarsaf** [comment, marked as answer]
Okay already solved. I have mistakenly put `shuffle` to `True` on the test_data.

**Reply by giotine** [reply]
Thank you so much for that question. This was driving me insane for hours!

**Reply by shisirkha** [reply]
Yeah, thanks for the tip.

**Reply by macayu17** [reply]
Thank you so much. I was stuck because of it for 2 days!
